src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -733,9 +733,82 @@ static T[] AllocateNewUninitializedArray(int length, bool pinned)
         [MethodImpl(MethodImplOptions.InternalCall)]
         private static extern long _GetTotalPauseDuration();
 
+        /// <summary>
+        /// Gets the total amount of time paused in GC since the beginning of the process.
+        /// </summary>
+        /// <returns> The total amount of time paused in GC since the beginning of the process.</returns>
         public static TimeSpan GetTotalPauseDuration()
         {
             return new TimeSpan(_GetTotalPauseDuration());
         }
+
+        internal struct GCConfigurationContext
+        {
+            internal Dictionary<string, object> Configurations;
+        }
+
+        [UnmanagedCallersOnly]
+        private static unsafe void Callback(void* configurationContext, void* name, void* publicKey, GCConfigurationType type, long data)
+        {
+            // If the public key is null, it means that the corresponding configuration isn't publicly available
+            // and therefore, we shouldn't add it to the configuration dictionary to return to the user.
+            if (publicKey == null)
+            {
+                return;
+            }
+
+            Debug.Assert(name != null);
+            Debug.Assert(configurationContext != null);
+
+            ref GCConfigurationContext context = ref Unsafe.As<byte, GCConfigurationContext>(ref *(byte*)configurationContext);
+            Debug.Assert(context.Configurations != null);
+            Dictionary<string, object> configurationDictionary = context.Configurations!;
+
+            string nameAsString = Marshal.PtrToStringUTF8((IntPtr)name)!;
+            switch (type)
+            {
+                case GCConfigurationType.Int64:
+                    configurationDictionary[nameAsString] = data;
+                    break;
+
+                case GCConfigurationType.StringUtf8:
+                    {
+                        string? dataAsString = Marshal.PtrToStringUTF8((IntPtr)data);
+                        configurationDictionary![nameAsString] = dataAsString ?? string.Empty;
```

Comment:
Is the `!` needed for the indexer? I don't see it above and I'm not sure why it is needed.